File: src/games/individual/longestword.ts
import { GameSession, AiResponse } from '../../types';
import { BaseGame } from '../base';
import { AiService } from '../../ai/api';
import { SessionManager } from '../../session/manager';
import { PointsSystem } from '../../points/system';
import { GroupManager } from '../../group/manager';
import { LeaderboardManager } from '../../leaderboard/manager';
import { KvStore } from '../../db/kv';
import { DbStore } from '../../db/d1';
import { TelegramApi } from '../../telegram';
import { Env } from '../../types';

export class LongestWordGame extends BaseGame {
  private topics = ['الحيوانات', 'النباتات', 'المدن العربية', 'المهن', 'الألوان', 'الأطعمة', 'المشروبات', 'الرياضات', 'الأدوات', 'المشاعر', 'العلوم', 'الفنون'];

  constructor(
    ai: AiService, sessionManager: SessionManager, pointsSystem: PointsSystem,
    groupManager: GroupManager, leaderboardManager: LeaderboardManager,
    kv: KvStore, db: DbStore, telegram: TelegramApi, env: Env
  ) { super(ai, sessionManager, pointsSystem, groupManager, leaderboardManager, kv, db, telegram, env); }

  getGameId(): string { return 'longestword'; }

  async generateQuestion(session: GameSession, round: number): Promise<AiResponse> {
    return {
      success: true, question: this.topics[round % this.topics.length], options: [], correctAnswer: '',
      storyBranch: [], choices: [], hint: '', explanation: '',
      metadata: {},
    };
  }

  async checkAnswer(session: GameSession, userId: number, answer: string): Promise<boolean> {
    const q = session.currentQuestion;
    if (!q) return false;
    const word = answer.trim();
    if (word.length < 3) return false;
    const prev = session.roundData?.answers?.[userId];
    if (prev) return word.length > prev.length;
    return true;
  }

  renderQuestion(session: GameSession, q: AiResponse): string {
    return `📏 الموضوع: ${q.question}\n\nاكتب أطول كلمة ممكنة متعلقة بالموضوع:`;
  }

  renderRoundStart(session: GameSession): string {
    return '🎯 لعبة أطول كلمة!\n\nسيتم عرض موضوع وعليك كتابة أطول كلمة ممكنة متعلقة به.';
  }
}

File: python_bot/games/longestword.py
#!/usr/bin/env python3
"""أطول كلمة - اكتب أطول كلمة ضمن الموضوع"""
import random
from games.base import BaseGame

class LongestWordGame(BaseGame):
    def get_game_id(self):
        return "longestword"
    def generate_question(self, session):
        topics = ["الحيوانات", "النباتات", "المدن العربية", "المهن", "الألوان", "الأطعمة", "المشروبات", "الرياضات"]
        return {"question": random.choice(topics), "correct_answer": ""}
    def check_answer(self, session, user_id, answer):
        word = answer.strip()
        if len(word) < 3:
            return False
        scores = session['scores']
        sid = str(user_id)
        prev = scores.get(sid, 0)
        if len(word) > prev:
            scores[sid] = len(word)
            return True
        return False
    def render_question(self, session, question):
        return f"📏 الموضوع: {question['question']}\n\nاكتب أطول كلمة ممكنة متعلقة بالموضوع:"